src/traversals/bfs.jl

```diff
@@ -116,149 +15,43 @@ function tree(parents::AbstractVector{T}) where T<:Integer
-        s::Integer;
-        vertexcolormap = Dict{T,Int}(),
-        queue = Vector{T}()) where T<:Integer
-    # this version of bfs_tree! allows one to reuse the memory necessary to compute the tree
-    # the output is stored in the visitor.tree array whose entries are the vertex id of the
-    # parent of the index. This function checks if the scratch space is too small for the graph.
-    # and throws an error if it is too small.
-    # the source is represented in the output by a fixed point v[root] == root.
-    # this function is considered a performant version of bfs_tree for useful when the parent
-    # array is more helpful than a DiGraph struct, or when performance is critical.
-    nvg = nv(g)
-    length(visitor.tree) >= nvg || error("visitor.tree too small for graph")
-    visitor.tree[s] = s
-    traverse_graph!(g, BreadthFirst(), s, visitor; vertexcolormap=vertexcolormap, queue=queue)
-end
-
 """
-    bfs_tree(g, s)
+    bfs_parents(g, s)
 
-Provide a breadth-first traversal of the graph `g` starting with source vertex `s`,
-and return a directed acyclic graph of vertices in the order they were discovered.
-
-### Implementation Notes
-This function is a high level wrapper around [`bfs_tree!`](@ref); use that function for more performance.
-"""
-function bfs_tree(g::AbstractGraph, s::Integer)
-    nvg = nv(g)
-    visitor = TreeBFSVisitorVector(nvg)
-    bfs_tree!(visitor, g, s)
-    return tree(visitor)
-end
-
-############################################
-# Connected Components with BFS            #
-############################################
+Perform a breadth-first search of graph `g` starting from vertex `s`.
+Return a vector of parent vertices indexed by vertex.
 """
-    ComponentVisitorVector{T}
-
-A type of `AbstractGraphVisitor` that represents connected components with
-BFS starting from a given seed.
-"""
-mutable struct ComponentVisitorVector{T<:Integer} <: AbstractGraphVisitor
-    labels::Vector{T}
-    seed::T
-end
-
-function examine_neighbor!(visitor::ComponentVisitorVector, u::Integer, v::Integer,
-                            ucolor::Int, vcolor::Int, ecolor::Int)
-    if u != v && vcolor == 0
-        visitor.labels[v] = visitor.seed
-    end
-    return true
-end
-
-############################################
-# Test graph for bipartiteness             #
-############################################
-mutable struct BipartiteVisitor <: AbstractGraphVisitor
-    bipartitemap::Vector{UInt8}
-    is_bipartite::Bool
-end
-
-BipartiteVisitor(n::Integer) = BipartiteVisitor(zeros(UInt8, n), true)
-
-function examine_neighbor!(visitor::BipartiteVisitor, u::Integer, v::Integer,
-        ucolor::Int, vcolor::Int, ecolor::Int)
-    if vcolor == 0
-        visitor.bipartitemap[v] = (visitor.bipartitemap[u] == 1) ? 2 : 1
-    else
-        if visitor.bipartitemap[v] == visitor.bipartitemap[u]
-            visitor.is_bipartite = false
+function bfs_parents(g::AbstractGraph, s::Integer)
+    T = eltype(g)
```

Comment:
I think this `parents` refers to the fact that the returned data structure is a parents array. Not that it is with the out edges.